### PR TITLE
mrd-892 Added domain event for recommendation started.

### PR DESCRIPTION
**Pull request by bills7281**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt [hunk 1]
```diff
@@ -355,6 +360,21 @@ internal class RecommendationService(
     }
   }
 
+  private fun sendRecommendationStartedEvent(recommendationId: Long) {
+    val crn = recommendationRepository.findById(recommendationId).map { it.data.crn }.get()
+    val payload = MrdEvent(
+      message = MrdEventMessageBody(
+        eventType = "RECOMMENDATION_STARTED",
+        version = 1,
+        description = "Recommendation started",
```
Comment (jackmaloney) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt:369]: Think this needs to be "Recommendation started (recall or no recall)" to match the contract in MRD-921

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt [hunk 2]
```diff
@@ -355,6 +360,21 @@ internal class RecommendationService(
     }
   }
 
+  private fun sendRecommendationStartedEvent(recommendationId: Long) {
+    val crn = recommendationRepository.findById(recommendationId).map { it.data.crn }.get()
+    val payload = MrdEvent(
+      message = MrdEventMessageBody(
+        eventType = "RECOMMENDATION_STARTED",
```
Comment (jackmaloney) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt:367]: This doesn't match MRD-921

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt (hunk 1 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt]: Corrected

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt (hunk 2 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt]: It should all be ok now :)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt [hunk 3]
```diff
@@ -355,6 +362,25 @@ internal class RecommendationService(
...
+    val crn = recommendationRepository.findById(recommendationId).map { it.data.crn }.get()
+    val payload = MrdEvent(
+      message = MrdEventMessageBody(
+        eventType = "prison-recall.recommendation.started",
+        version = 1,
+        description = "Recommendation started (recall or no recall)",
+        occurredAt = LocalDateTime.now(),
+        detailUrl = "", // TODO TBD
+        personReference = PersonReference(listOf(TypeValue(type = "CRN", value = crn))),
+        additionalInformation = AdditionalInformation(recommendationUrl = "http://mrd.case.crn/overview")
```
Comment (jackmaloney) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt:375]: That was just an example URL for the ticket - it needs to be the correct URL to the overview tab for the case that has had the recommendation started.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt [hunk 4]
```diff
@@ -355,6 +362,25 @@ internal class RecommendationService(
...
+      ),
+      signingCertURL = "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-0000000000000000000000.pem",
```
Comment (jackmaloney) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt:377]: I don't think you need to add this, when I spoke to Matt about it the other day he said: "The rest of the wrapper attributes will be added by SNS automatically including the subscribeUrl and signingCertURL fields". 
So you should be good to remove these 2 fields.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt (hunk 3 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt]: hardcoded to prod for now, but I have asked Matt R for some help on making it dynamic to detect the host domain of whichever environment is in use

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**